packages/connection-info/src/connection-secrets.ts

```diff
@@ -190,10 +181,75 @@ export function extractSecrets(connectionInfo: Readonly<ConnectionInfo>): {
   return { connectionInfo: connectionInfoWithoutSecrets, secrets };
 }
 
-function omitPropertiesWhoseValuesAreEmptyObjects<
-  T extends Record<string, Record<string, unknown>>
->(obj: T): Partial<T> {
+function omitPropertiesWhoseValuesAreEmptyObjects<T extends Document>(obj: T) {
   return Object.fromEntries(
     Object.entries(obj).filter(([, value]) => Object.keys(value).length > 0)
-  ) as Partial<T>;
+  );
+}
+
+function extractAutoEncryptionSecrets(data: Document) {
+  function flattenObject(
+    obj: Document,
+    path = '',
+    result: Document = {}
+  ): Record<string, unknown> {
+    _.forEach(obj, (value, key) => {
+      key = key.replaceAll('.', '\\.');
```

Comment:
This probably works out okay because our keys do not contain literal `\.` sequences, but this code overall isn't the easiest to read – I don't think there's anything requiring us to stick to a paradigm where we specify the set of paths that we want to redact as a list of strings, manually redacting these objects the way we do for other sub-properties of `ConnectionOptions` is probably just fine too and might end up being both more readable and more type-safe